datarepo-clienttests/src/main/java/testscripts/SoftDeleteDataset.java

```diff
@@ -0,0 +1,215 @@
+            repositoryApi, ingestFileJobResponse, BulkLoadArrayResultModel.class);
+    String fileId = bulkLoadArrayResultModel.getLoadFileResults().get(0).getFileId();
+
+    // ingest the tabular data from the JSON file we just generated
+    // generate a JSON file with the fileref
+    String jsonLine =
+        "{\"VCF_File_Name\":\"name1\", \"Description\":\"description1\", \"VCF_File_Ref\":\""
+            + fileId
+            + "\"}\n";
+    byte[] fileRefBytes = jsonLine.getBytes(StandardCharsets.UTF_8);
+    String jsonFileName = FileUtils.randomizeName("this-better-pass") + ".json";
+    String dirInCloud = "scratch/softDel";
+    String fileRefName = dirInCloud + "/" + jsonFileName;
+    String gsPath = FileUtils.createGsPath(fileRefBytes, fileRefName, testConfigGetIngestbucket);
+
+    IngestRequestModel ingestRequest =
+        new IngestRequestModel()
+            .format(IngestRequestModel.FormatEnum.JSON)
+            .table("vcf_file")
+            .path(gsPath);
+    JobModel ingestTabularDataJobResponse =
+        repositoryApi.ingestDataset(datasetSummaryModel.getId(), ingestRequest);
+
+    ingestTabularDataJobResponse =
+        DataRepoUtils.waitForJobToFinish(repositoryApi, ingestTabularDataJobResponse);
+    IngestResponseModel ingestResponse =
+        DataRepoUtils.expectJobSuccess(
+            repositoryApi, ingestTabularDataJobResponse, IngestResponseModel.class);
+    logger.info("Successfully loaded data into dataset: {}", ingestResponse.getDataset());
+
+    String datasetId = datasetSummaryModel.getId(); // TODO should these be in utils?
+    DatasetModel datasetModel = repositoryApi.retrieveDataset(datasetId);
+    String dataProject = datasetModel.getDataProject();
+    String tableName = datasetModel.getSchema().getTables().get(0).getName();
+
+    // get row ids for table  TODO combine SQL query w queryForFileRefs in DRSLookup
+    String tableRef =
+        String.format(
+            "`%s.%s.%s`",
+            dataProject,
+            "datarepo_" + datasetModel.getName(), // TODO const ni BQ utils
+            tableName);
+    String sqlQuery =
+        String.format(
+            "SELECT %s FROM %s LIMIT %s",
+            "datarepo_row_id", tableRef, 1L); // TODO what is the limit?
+    TableResult result = BigQueryUtils.queryBigQuery(dataProject, sqlQuery);
+    List<String> rowIds =
+        StreamSupport.stream(result.getValues().spliterator(), false)
+            .map(fieldValues -> fieldValues.get(0).getStringValue())
+            .collect(Collectors.toList());
+
+    logger.info("Successfully retrieved row ids for table: {}", tableName);
+
+    // write to GCS
+    String csvFileName = FileUtils.randomizeName("this-too-better-pass") + ".csv";
+    String csvFileRefName = dirInCloud + "/" + csvFileName;
+    String gcsPath = FileUtils.createGcsPath(rowIds, csvFileRefName, testConfigGetIngestbucket);
+
+    // build the deletion request with pointers to the file with row ids to soft delete
```

Comment:
I think this file with the row ids to soft delete contains all the row ids in the table -- right? If so, then that means we could only run the user journey below once? If I've got that right, then I think this is a good test for soft deletes, but perhaps a slightly restructured version could be better for testing scaling of soft deletes. To test scaling (i.e. how many soft delete requests can we handle at a time) perhaps we could break this up to one soft delete request per row id? Then each user journey thread would try and soft delete one row id.

Reply:
follow up pr with this additional test / work to come!